test/client/client_test.go

```diff
@@ -0,0 +1,93 @@
+package client_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/cluster-logging-operator/test"
+	. "github.com/openshift/cluster-logging-operator/test/client"
+	. "github.com/openshift/cluster-logging-operator/test/matchers"
+	"github.com/openshift/cluster-logging-operator/test/object"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+var _ = Describe("Client", func() {
+	var (
+		c    *Client
+		ns   *corev1.Namespace
+		data map[string]string
+	)
+
+	BeforeEach(func() {
+		c = Get()
+		ns = c.CreateUniqueNS()
+		test.Infof("test namespace: %v", ns.Name)
+		data = map[string]string{"a": "b"}
+	})
+
+	AfterEach(func() {
+		test.UnlessFailed(func() { ExpectOK(c.Remove(ns)) })
```

Comment:
Does this imply that we keep the namespace if a failure occurs? Is this required for a test must-gather later?

Reply:
It does, that's something to be considered. When I'm running tests one-by-one, it's much easier if the actual namespace is still around. When we're running on CI I'm not sure if that's helpful or not. We might want to configure this based on whether we're in CI or not, or maybe it's just OK to leak these resources since they don't interfere with other tests and can be cleaned up with an `oc delete -l test-client` We'll see how it works in practice.